src/singletons/StreamerMode.cpp

```diff
@@ -0,0 +1,253 @@
+        u"XSplit.Core.exe"_s, u"TwitchStudio.exe"_s, u"vMix64.exe"_s,
+    };
+#else
+    static QStringList bins = {
+        u"obs"_s,
+        u"Twitch Studio"_s,
+        u"Streamlabs Desktop"_s,
+    };
+#endif
+    return bins;
+}
+
+bool isBroadcasterSoftwareActive()
+{
+#if defined(Q_OS_LINUX) || defined(Q_OS_MACOS)
+    QProcess p;
+    p.start("pgrep", {"-x", broadcastingBinaries().join("|")},
+            QIODevice::NotOpen);
+
+    if (p.waitForFinished(1000) && p.exitStatus() == QProcess::NormalExit)
+    {
+        return (p.exitCode() == 0);
+    }
+
+    // Fallback to false and showing a warning
+
+    static bool shouldShowWarning = true;
+    if (shouldShowWarning)
+    {
+        shouldShowWarning = false;
+
+        postToThread([] {
+            getApp()->twitch->addGlobalSystemMessage(
+                "Streamer Mode is set to Automatic, but pgrep is missing. "
+                "Install it to fix the issue or set Streamer Mode to "
+                "Enabled or Disabled in the Settings.");
+        });
+    }
+
+    qCWarning(chatterinoStreamerMode) << "pgrep execution timed out!";
+    return false;
+#elif defined(Q_OS_WIN)  // TODO(C++ 23): use elifdef
+    if (!IsWindowsVistaOrGreater())
+    {
+        return false;
+    }
+
+    WTS_PROCESS_INFO *pProcessInfo = nullptr;
+    DWORD dwProcCount = 0;
+
+    if (WTSEnumerateProcesses(WTS_CURRENT_SERVER_HANDLE, 0, 1, &pProcessInfo,
+                              &dwProcCount))
+    {
+        //Go through all processes retrieved
+        for (DWORD i = 0; i < dwProcCount; i++)
+        {
+            QStringView processName(pProcessInfo[i].pProcessName);
+
+            if (broadcastingBinaries().contains(processName))
+            {
+                WTSFreeMemory(pProcessInfo);
+                return true;
+            }
+        }
+    }
+
+    if (pProcessInfo)
+    {
+        WTSFreeMemory(pProcessInfo);
+    }
+
+#else
+#    warning Unsupported OS: Broadcasting software can't be detected
+#endif
+    return false;
+}
+
+class SingleConnection
+{
+public:
+    // NOLINTNEXTLINE(readability-identifier-naming)
+    void emplace_back(auto &&conn)
+    {
+        this->conn_ = std::forward<decltype(conn)>(conn);
+    }
+
+private:
+    std::unique_ptr<pajlada::Signals::ScopedConnection> conn_;
+};
+
+}  // namespace
+
+namespace chatterino {
+
+using namespace std::chrono_literals;
+
+class StreamerModePrivate
+{
+public:
+    StreamerModePrivate(StreamerMode *parent_);
+
+    [[nodiscard]] bool isEnabled() const;
+
+private:
+    void settingChanged(StreamerModeSetting value);
+    void setEnabled(bool enabled);
+
+    void check();
+
+    StreamerMode *parent_;
+    SingleConnection settingConnection_;
+
+    QTimer timer_;
+    QThread thread_;
+
+    std::atomic_bool enabled_ = false;
+    mutable std::atomic_uint8_t timeouts_ = 0;
+    StreamerModeSetting prevValue_ = StreamerModeSetting::Disabled;
```

Comment:
called it Setting instead of Value
Calling it `prev` also implies that this will contain the setting as-it-were prior to the current value, which I don't think is what it stores